common/crypto_utils/avalon_crypto_utils/signature.py

```diff
@@ -323,35 +323,69 @@ def generate_client_signature(
         return input_json_str, SignatureStatus.PASSED
 
 # -----------------------------------------------------------------------------
-    def verify_signature(self, input_json, verification_key):
+    def verify_signature(self, wo_response, verification_key,
+                         requester_nonce):
         """
         Function to verify the signature received from the enclave
         Parameters:
-            - input_json is dictionary contains payload returned by the
+            @param wo_response - dictionary contains payload returned by the
               Worker Service in response to successful workorder submit request
               as per Trusted Compute EEA API 6.1.2 Work Order Result Payload
-            - verification_key is ECDSA/SECP256K1 public key used to verify
+            @param verification_key - ECDSA/SECP256K1 public key used to verify
               signatures created by the Enclave.
+            @param requester_nonce - requester generated nonce passed in work
+            order request. Required in 2 step verification.
         Returns enum type SignatureStatus
         """
-
-        input_json_params = input_json
-
-        nonce = (input_json_params['workerNonce']).encode('UTF-8')
-        signature = input_json_params['workerSignature']
-
+        v_key = None
+        # if verification_key present in work order response
+        # then do 2 step verification
+        # step1 - The verification key signature from the
+        # response is verified using worker’s public verification key
+        # (aka KME's verification key)
+        if "extVerificationKey" in wo_response:
+            decrypted_v_key = crypto.base64_to_byte_array(
+                wo_response["extVerificationKey"])
+            v_key = crypto.byte_array_to_string(decrypted_v_key)
```

Comment:
extVerificationKeySignature is computed on the hash of the concatenation of b64 encoded verification key and requester nonce hex string.

Reply:
pk. Then we should use extVerificationKey instead of vkey. 
Is extVerificationKey  in base64 encoded format.?
If so,  concat_string = extVerificationKey  + requester_nonce ?

Reply:
Hash is calculated on extVerificationKey + requester_nonce